python/cuml/test/test_pickle.py

```diff
@@ -18,50 +18,44 @@
 import pickle
 import pytest
 
-from cuml.test.utils import array_equal, unit_param, stress_param
+from cuml.test.utils import array_equal, unit_param, stress_param, \
+    ModuleConfig
 from cuml.test.test_svm import compare_svm
 from sklearn.base import clone
-from sklearn.datasets import load_iris, make_classification,\
-    make_regression
+from sklearn.datasets import load_iris, make_classification, make_regression
 from sklearn.manifold.t_sne import trustworthiness
 from sklearn.model_selection import train_test_split
 
-regression_models = {
-    "LinearRegression": lambda fit_intercept=True: cuml.LinearRegression(
-        fit_intercept=fit_intercept),
-    "LogisticRegression": lambda fit_intercept=True: cuml.LogisticRegression(
-        fit_intercept=fit_intercept),
-    "Lasso": lambda fit_intercept=True: cuml.Lasso(
-        fit_intercept=fit_intercept),
-    "Ridge": lambda fit_intercept=True: cuml.Ridge(
-        fit_intercept=fit_intercept),
-    "ElasticNet": lambda fit_intercept=True: cuml.ElasticNet(
-        fit_intercept=fit_intercept)
-}
-
-solver_models = {
-    "CD": lambda: cuml.CD(),
-    "SGD": lambda: cuml.SGD(eta0=0.005),
-    "QN": lambda: cuml.QN(loss="softmax")
-}
-
-cluster_models = {
-    "KMeans": lambda: cuml.KMeans()
-}
-
-decomposition_models = {
-    "PCA": lambda: cuml.PCA(),
-    "TruncatedSVD": lambda: cuml.TruncatedSVD(),
-}
-
-decomposition_models_xfail = {
-    "GaussianRandomProjection": lambda: cuml.GaussianRandomProjection(),
-    "SparseRandomProjection": lambda: cuml.SparseRandomProjection()
-}
-
-neighbor_models = {
-    "NearestNeighbors": lambda: cuml.NearestNeighbors()
-}
+regression_config = ModuleConfig(module=cuml.linear_model)
+regression_models = regression_config.get_models()
+
+solver_config = ModuleConfig(
+    module=cuml.solvers,
+    custom_constructors={
+        "SGD": lambda: cuml.SGD(eta0=0.005),
+        "QN": lambda: cuml.QN(loss="softmax")
```

Comment:
any reason to not just use default parameters? Just wondering why this specific values

Reply:
So I have just tried both with default arguments, for SGD it works with defaults so I guess we can use that instead. For QN though there is those checks in the fit method that breaks with default arguments:
```
if self.loss_type != 2 and self.num_classes > 2:
            raise ValueError("Only softmax (multinomial) loss supports more"
                             "than 2 classes.")

if self.loss_type == 2 and self.num_classes <= 2:
            raise ValueError("Only softmax (multinomial) loss supports more"
                             "than 2 classes.")
```

Reply:
Can you create a github issue describing this problem?